geode-core/src/main/java/org/apache/geode/internal/cache/TXEntryState.java

```diff
@@ -372,7 +375,7 @@ public void updateForWrite(final int modNum) {
    * @since GemFire 5.1
    */
   public boolean isDirty() {
-    return DETECT_READ_CONFLICTS || this.dirty;
+    return DETECT_READ_CONFLICTS || detectReadConflicts || this.dirty;
```

Comment:
You could add a method on TXRegionState "boolean isDetectReadConflicts() { return DETECT_READ_CONFLICTS || detectReadConflicts;}" and move the sys prop constant to that class.